### ISSUE TITLE
Legacy boot fail when no primary partition is set bootable[FIRMWARE BUG]

### ISSUE THREAD

**Issue by jphilip09** [slacka/WoeUSB#97]
WinUSB was able to write successfully to the usb drive, but it did not set the partition to bootable. Setting the bootable flag on the partition using fdisk, I was able to boot with the usb drive.

```
$ sudo winusb --format Win10_1607_EnglishInternational_x64.iso /dev/sdc
Formatting device...
Wait 3 seconds for block device nodes to populate...
mkfs.fat 3.0.28 (2015-05-16)
mkfs.fat: warning - lowercase labels might not work properly with DOS or Windows
Mounting...
Copying...
Installing grub...
Installing for i386-pc platform.
grub-install: warning: Attempting to install GRUB to a disk with multiple partition labels.  This is not supported yet..
grub-install: warning: Embedding is not possible.  GRUB can only be installed in this setup by using blocklists.  However, b
locklists are UNRELIABLE and their use is discouraged..
Installation finished. No error reported.
Installing grub.cfg...
Exiting...
Syncing...
Cleaning...
Unmounting and removing '/media/winusb_iso_1495012212_8768'...
Unmounting and removing '/media/winusb_target_1495012212_8768'...

$ sudo fdisk -l /dev/sdc
Disk /dev/sdc: 7.5 GiB, 8000110592 bytes, 15625216 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x5dc04c8c

Device     Boot Start      End  Sectors  Size Id Type
/dev/sdc1        8192 15625215 15617024  7.5G  c W95 FAT32 (LBA)
```

**Comment by Lin-Buo-Ren** [comment]
GRUB PC doesn't really rely on partition's boot flag for it's booting so it
should be a different kind of issue

2017年5月17日 下午5:51，"Philip Joseph" <notifications@github.com>寫道：

**Comment by Lin-Buo-Ren** [comment]
NOTE: the GRUB's warning shouldn't happen in regular run, atill need to investigate though

You may try to use `wipefs` to remove all partition trail data and run WinUSB again to verify the issue

**Comment by Lin-Buo-Ren** [comment]
Refer: <https://unix.stackexchange.com/questions/23560/what-is-the-bootable-flag-option-when-installing-a-distro>

**Comment by jphilip09** [comment]
The bootable flag is required when booting in the legacy mode. If I use UEFI, it is booting correctly.

**Comment by Lin-Buo-Ren** [comment]
@jphilip09 You have wrong assumption.
Currently the only major bootloader that requires partition boot flag is Windows NTLDR and BOOTMGR, which MBR code check the partitions' boot flag, and jump to the VBR of the partition which boot flag is toggled.

GNU GRUB in PC paradigm does NOT use this mechanism, instead it's MBR code jumps to the embedded code in the post-MBR gap, then directly loads it's other parts in the /boot partition.

**Comment by Lin-Buo-Ren** [comment]
Refer:

* [GNU GRUB Manual > GRUB image files](https://www.gnu.org/software/grub/manual/grub.html#Images)
* [GNU GRUB Manual > Installation > BIOS installation](https://www.gnu.org/software/grub/manual/grub.html#BIOS-installation)

**Comment by Lin-Buo-Ren** [comment]
This issue is pretty much a duplicate of #61, please check for possible workarounds because the cause is currently unknown

**Comment by JonnyTech** [comment]
I am still seeing this problem with v2.3.0 and have to manually set the partition bootable with fdisk

**Comment by Lin-Buo-Ren** [comment]
Unfortunately I still can't reproduce this issue on Oracle VirtualBox and other machines, I assume it is rather a firmware bug than a bug in WoeUSB

**Comment by Lin-Buo-Ren** [comment]
Refer the second answer by Thomas Lindner in [boot - What is the "Bootable flag" option when installing a distro? - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/23560/what-is-the-bootable-flag-option-when-installing-a-distro), this is a firmware bug rather than bug in WoeUSB.  We will still deal with this, however in a workaround form rather than a fix.

**Comment by Lin-Buo-Ren** [comment]
To @jphilip09 , and other that affected by this issue: 

An *optional* workaround has been implemented in WoeUSB v2.5.0, please comment if you still reproduce this issue using the workaround.
